### PR TITLE
RELATED: RAIL-4495 Connecting attributes loading

### PR DESCRIPTION
**Pull request by plavemar**
- remove connectingAttributesMatrix and all its related types and functions
- introduce query for fetching connecting attributes from backend on demand
- optimize useParentConfiguration hook 

JIRA: RAIL-4495

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/parentFilters/ParentFiltersListItem.tsx [hunk 1]
```diff
@@ -56,17 +55,14 @@ export const ParentFiltersListItem: React.FC<IConfigurationParentItemProps> = (p
...
-        );
+        const connectingAttributeRefs = connectingAttributes.map((attr) => attr.ref);
+        onClick(localIdentifier, !isSelected, connectingAttributeRefs);
     }, [isSelected, localIdentifier, onClick, connectingAttributes]);
 
     if (isDisabled) {
         return (
             <ParentFiltersDisabledItem
-                hasConnectingAttributes={!!connectingAttributes.length}
+                hasConnectingAttributes={true}
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/parentFilters/ParentFiltersListItem.tsx:65]: Is this intentional? The previous `!!connectingAttributes.length` seemed reasonable.

File: libs/sdk-ui-dashboard/src/model/queries/connectingAttributes.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import { ObjRef } from "@gooddata/sdk-model";
+import { IDashboardQuery } from "./base";
+
+/**
+ * @internal
+ */
+export interface QueryConnectingAttributes extends IDashboardQuery {
+    type: "GDC.DASH/QUERY.CONNECTING.ATTRIBUTES";
+    payload: {
+        readonly refs: ObjRef[][];
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/queries/connectingAttributes.ts:12]: Should this be 
```ts
readonly refs: [ObjRef, ObjRef][]
```
? So it enforces pairs?

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9882/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11789/ : FAILURE in 10m 12s

File: libs/sdk-ui-dashboard/src/model/queries/connectingAttributes.ts (hunk 2 above)
Comment (plavemar) [reply libs/sdk-ui-dashboard/src/model/queries/connectingAttributes.ts:12]: type updated

File: libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/parentFilters/ParentFiltersListItem.tsx (hunk 1 above)
Comment (plavemar) [reply libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/parentFilters/ParentFiltersListItem.tsx:65]: forgotten workaround for implementation steps. Moved back to length of connecting attributes

File: libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/hooks/useConnectingAttributes.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+} from "../../../../../../model";
+
+/**
+ * @internal
+ */
+export function useConnectingAttributes(
+    currentFilterDisplayForm: ObjRef,
+    neighborFiltersDisplayForms: ObjRef[],
+) {
+    const pairs: ObjRef[][] = useMemo(
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/hooks/useConnectingAttributes.ts:18]: The type here should also be the pairs

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9886/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11792/ : FAILURE in 7m 20s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9887/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11793/ : SUCCESS in 14m 51s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3008

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2818/ : SUCCESS in 2m 13s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3122/ : SUCCESS in 33s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1963/ : SUCCESS in 24m 13s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11797/ : SUCCESS in 24m 50s

**Pull request merged**